app/schema.json

```diff
@@ -4,31 +4,105 @@
   "title": "HiGlass viewconf",
   "type": "object",
   "additionalProperties": false,
+  "required": [],
   "properties": {
     "editable": { "type": "boolean" },
     "zoomFixed": { "type": "boolean" },
     "exportViewUrl": { "type": "string" },
     "trackSourceServers": {
       "type": "array",
-      "items": { "type": "string" }
+      "items": { "type": "string" },
+      "minLength": 1
     },
-    "zoomLocks": { "$ref": "#/definitions/locks" },
-    "locationLocks": { "$ref": "#/definitions/locks" },
-    "valueScaleLocks": { "$ref": "#/definitions/locks" },
+    "zoomLocks": { "$ref": "#/definitions/locks/genericLocks" },
+    "locationLocks": { "$ref": "#/definitions/locks/genericLocks" },
+    "valueScaleLocks": { "$ref": "#/definitions/locks/valueScaleLocks" },
     "views": {
       "type": "array",
-      "items": { "$ref": "#/definitions/view" }
+      "items": { "$ref": "#/definitions/view" },
+      "minLength": 1
     }
   },
   "definitions": {
 
     "locks": {
-      "type": "object",
-      "additionalProperties": false,
-      "required": ["locksByViewUid", "locksDict"],
-      "properties": {
-        "locksByViewUid": { "type": "object" },
-        "locksDict": { "type": "object" }
+      "string22": {
```

Comment:
This isn't necessary. I've been using 22 character slugid's but as long as the identifiers are unique, there's no reason why they need to be constrained to 22 character strings.